Create-detrend-object

jExbrayat opened the pull request:
Create detrend.py file defining the detrend class. 
Create a temporary demo notebook for easier visualisation of the detrend class' aim.

jExbrayat committed:
feat: create detrend class

jExbrayat committed:
feat: add return to predict function

predict now returns the fitted trend array

jExbrayat committed:
feat: create demo notebook for detrend class

jExbrayat committed:
feat: add fancy plot function to detrend class

jExbrayat committed:
feat: add optional xticklabels for fancy_plot function

jExbrayat committed:
refactor: add method's name in fancy plot title

jExbrayat committed:
fix: display true method's name in fancy plot title

Before the fix, the function string object was plotted

jExbrayat committed:
refactor: create proper demo and clean notebook

jExbrayat committed:
docs: clarify what is self.method with commentary

jExbrayat committed:
feat: add parameters plot in graph's title for polynomial method

jExbrayat committed:
refactor: add parameters plot compatibility to linear regression

jExbrayat committed:
refactor: generalize parameters plot in title for all methods

jExbrayat committed:
chore: execute demo notebook with new parameters plot

Ab2nour commented on src/detrend.py:
```diff
@@ -0,0 +1,211 @@
+
+        return y_interpolate, parameters_output
+
+    def __init__(
```
init should be the first method of the class

Ab2nour commented on src/detrend.py:
```diff
@@ -0,0 +1,211 @@
+        self.y_predict = self.y_original - self.fitted_values
+        return self.y_predict
+
+    def fancy_plot(self, xticklabels=None) -> None:
```
add type hint for xticklabels

Ab2nour replied:
ok :)

jExbrayat commented:
>  ok boomer

jExbrayat commented:
> init should be the first method of the class 

sorry boomer

jExbrayat committed:
refactor: add type hint for fancy_plot method

jExbrayat committed:
refactor: add type hint for methods variable

jExbrayat committed:
refactor: put init method at beginning of class

Ab2nour commented on src/detrend.py:
```diff
@@ -0,0 +1,212 @@
+import matplotlib.pyplot as plt
+
+
+class detrend:
```
class names should be PascalCase

jExbrayat committed:
refactor: put class name in PascalCase

jExbrayat committed:
refactor: update detrend class name

Ab2nour commented on src/detrend.py:
```diff
@@ -0,0 +1,212 @@
+        n_segments: int = 5,
+        window: int = 100,
+        alpha: float = 0.05,
+        bsplines_factors: tuple = (10, 3),
```
type hint: set the size of the tuple
for example, tuple of 2 ints should be type hinted this way:
```py
bsplines_factors: tuple[int, int] = (10, 3),
```

Ab2nour replied:
type hint: set the size of the tuple
for example, tuple of 2 ints should be type hinted this way:
```py
bsplines_factors: tuple[int, int] = (10, 3),
```

Ab2nour replied:
type hint: set the size of the tuple
for example, tuple of 2 ints should be type hinted this way:
```py
bsplines_factors: tuple[int, int] = (10, 3),
```

Ab2nour commented on src/detrend.py:
```diff
@@ -0,0 +1,212 @@
+        n_segments: int = 5,
+        window: int = 100,
+        alpha: float = 0.05,
+        bsplines_factors: tuple = (10, 3),
```
type hint: set the size of the tuple
for example, tuple of 2 ints should be type hinted this way:
```py
bsplines_factors: tuple[int, int] = (10, 3),
```

jExbrayat replied:
done

Ab2nour commented on src/detrend.py:
```diff
@@ -0,0 +1,212 @@
+        self.y_predict = self.y_original - self.fitted_values
+        return self.y_predict
+
+    def fancy_plot(self, xticklabels: pd.core.indexes.base.Index | None = None) -> None:
```
add docstring

jExbrayat replied:
done

Ab2nour commented on src/detrend.py:
```diff
@@ -0,0 +1,212 @@
+        }
+
+        # Define x and y
+        x = np.arange(len(y))  # time dummy, x
```
maybe x and y should instead be named time_dummy and price?

jExbrayat replied:
done

jExbrayat committed:
refactor: add type hint for bsplines_factors argument of init method

jExbrayat committed:
refactor: modify variables name for clarity

jExbrayat committed:
docs: add doc string for fancy_plot method

jExbrayat committed:
docs: delete useless commentary following variables names change

jExbrayat committed:
fix: correct old variable name

jExbrayat committed:
refactor: rename detrend.py to old_detrend.py

jExbrayat committed:
feat: create LinearReg class

jExbrayat committed:
refactor: create external _fancy_plot function

jExbrayat committed:
refactor: put self variable in _fancy_plot function's argument instead of string

jExbrayat committed:
feat: create PolynomialRegression class

jExbrayat committed:
feat: create LinearMA class

jExbrayat committed:
feat: create ExponentialMA class

jExbrayat committed:
feat: create BSplines class

jExbrayat committed:
refactor: delete useless imports

jExbrayat committed:
chore: delete old_detrend.py

jExbrayat committed:
refactor: adapt and clean demo notebook

jExbrayat committed:
docs: correct syntax

jExbrayat committed:
docs: correct syntax

jExbrayat committed:
feat: create detrend and write process script

jExbrayat committed:
refactor: delete .csv in stock name and add .csv at end of processed filename

jExbrayat committed:
refactor: execute script for polynomial regression model

jExbrayat committed:
feat: create demo notebook for detrended stock data

jExbrayat committed:
feat: plot candle chart of detrend data against original data

jExbrayat committed:
refactor: improve detrend classes and adapt demo detrend notebook

Add possibility to specify different time series for fit and predict as long as they are the same length.

Useful to detrend stock prices with several values at each time step (open, close, high, low).

jExbrayat committed:
refactor: fit only 1 time series per stock in write_detrend_data and correct filename

The stock data are now detrended using the close price (arbitrary). This method allows coherent candle stick data, takes less time to process and gives an equally correct result.

Processed filenames no longer contain an undesired dot after their stock name.

jExbrayat committed:
refactor: re execute write_detrend_data.py and demo notebook

jExbrayat committed:
feat/writing-detrended-data-to-csv

Create script to write results of detrended data in csv files.

jExbrayat committed:
refactor: change save directory for detrended files

jExbrayat committed:
fix: set right directory to import detrend module

jExbrayat committed:
refactor: reduce code length thanks to native option

jExbrayat committed:
feat: run write_detrend_data.py for LinearMA

jExbrayat committed:
docs: explain model_options format for directory creation

jExbrayat committed:
refactor/change-organization-of-processed-data

Refactor/change organization of processed data

jExbrayat committed:
feat: add seasonal plot

jExbrayat committed:
docs: update outline

jExbrayat committed:
feat: add periodogram

jExbrayat committed:
refactor: import detrend data and re-execute  notebook

jExbrayat committed:
feat: experiment seasonal regression

Ab2nour committed:
Merge pull request #17 from Ab2nour/refactor/separate-methods-in-classes

Refactor/separate methods in classes

Ab2nour commented:
I merged [refactor/separate-methods-in-classes](https://github.com/Ab2nour/stock-analysis/pull/13/commits/b1bf723bc6c4f434ac4cc2e397243bb674538299) into this branch.

This branch also looks good to me, you can merge it into main, if you are finished @jExbrayat .

jExbrayat committed:
Merge pull request #27 from Ab2nour:feat/seasonality-based-on-detrended-data

feat: seasonality-based-on-detrended-data

jExbrayat committed:
refactor: separate data into processed and raw folders

jExbrayat committed:
docs: create doc for detrend_data folder

jExbrayat committed:
docs: precise doc for raw_data folder

jExbrayat committed:
refactor: create docs folder for demo notebooks

jExbrayat committed:
refactor: create src/functions folder

jExbrayat committed:
refactor: create src/notebooks folder

jExbrayat committed:
refactor: create src/notebooks folder

jExbrayat committed:
refactor: create src/data_processing folder

jExbrayat committed:
Merge branches 'create-detrend-object' and 'create-detrend-object' of https://github.com/Ab2nour/stock-analysis into create-detrend-object

jExbrayat committed:
refactor: update data path

jExbrayat committed:
refactor: update data path

jExbrayat committed:
refactor: update data path

jExbrayat committed:
refactor: update data path

jExbrayat committed:
refactor: resolve NoModuleFound problem

jExbrayat committed:
refactor: clean and run demo notebook

jExbrayat committed:
refactor: clean and run demo notebook

jExbrayat committed:
docs: add decomposition.ipynb description in readme

jExbrayat committed:
docs: change title of readme for notebooks folder

jExbrayat committed:
docs: create readme for src/functions folder

jExbrayat committed:
docs: add readme for docs folder

jExbrayat committed:
refactor: move data processing script into functions folder

jExbrayat committed:
refactor: delete useless sys.path.append solution

jExbrayat committed:
Revert "refactor: delete useless sys.path.append solution"

This reverts commit 4fc9b258fd34023eb0784a94861e5571d2d1f43a.
